data-listener/src/main/kotlin/com/playkids/onboarding/data/listener/DataListener.kt

```diff
@@ -0,0 +1,51 @@
+package com.playkids.onboarding.data.listener
+
+import com.playkids.onboarding.sqs.SQSHandler
+import com.playkids.onboarding.sqs.SQSListener
+import kotlinx.coroutines.GlobalScope
+import kotlinx.coroutines.Job
+import kotlinx.coroutines.future.await
+import kotlinx.coroutines.launch
+import software.amazon.awssdk.services.sqs.SqsAsyncClient
+import software.amazon.awssdk.services.sqs.model.DeleteMessageRequest
+import software.amazon.awssdk.services.sqs.model.ReceiveMessageRequest
+import java.util.concurrent.atomic.AtomicBoolean
+
+class DataListener(
+    private val sqs: SqsAsyncClient,
+    private val queueUrl: String,
+    private val maxNumberOfMessages: Int,
+    private val messageHandler: SQSHandler
+): SQSListener {
+
+    private val isRunning = AtomicBoolean(false)
+
+    override fun stop() = isRunning.set(false)
+
+    override fun start(): Job = GlobalScope.launch {
+        isRunning.set(true)
+        do {
+            sqs.receiveMessage(
+                ReceiveMessageRequest.builder()
+                    .queueUrl(queueUrl)
+                    .maxNumberOfMessages(maxNumberOfMessages)
+                    .messageAttributeNames(listOf("entity", "operation"))
+                    .build()
+            )
+                .await()
+                .messages()
+                .forEach {
+                    messageHandler.handle(it.body(), it.messageAttributes().mapValues { entry -> entry.value.stringValue() })
```

Comment:
it would be a good idea to do a try catch here and delete message in finally block, but as you can guarantee `messageHandler.handle` won't throw Exceptions in your project it is ok